app/Controllers/Http/ProfileController.ts

```diff
@@ -0,0 +1,137 @@
+          throw new ResourceExistError("Mobile number already used");
+        }else{
+          //Create the profile and saved the data on database
+          const profile = await Profile.create({
+            name,
+            gender,
+            mobileNumber,
+            dateOfBirth,
+            userId: loginUserId,
+          });
+          response.created(profile);
+        }
+      }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  //For getting the profile of login user
+  public async getUserProfile({ response, auth }) {
+    try {
+      const loginUserId = auth.user.id;
+      const userProfile = await Profile.findByOrFail("userId", loginUserId);
+      const user = await User.findOrFail(loginUserId);
+      const userDetails = {
+        name: userProfile.name,
+        email: user.email,
+        gender: userProfile.gender,
+        dateOfBirth: new Date(userProfile.dateOfBirth).toLocaleDateString(),
+      };
+      response.ok(userDetails);
+    } catch (error) {
+      throw error;
+    }
+  }
+  //For updating the login user profile
+  public async updateUserProfile({ request, response, auth }) {
+    try {
+      const loginuserId = auth.user.id;
+      const exitsedProfile = await Profile.findByOrFail("userId", loginuserId);
+      const { name, gender, mobileNumber, dateOfBirth } =
+        await request.validate(UserProfileValidator);
+      const age = moment().diff(new Date(dateOfBirth), "years");
+
+      if (name === undefined || mobileNumber === undefined) {
+        const fieldName = !name ? "name" : "mobileNumber";
+        throw new UndefinedFieldError(`${name} required`, fieldName); 
+      }else if(age <18 || age >80){
+        throw new InvalidDataError("Age should be between 18-80");
+      }else if(exitsedProfile.mobileNumber !== mobileNumber){
+        //Checking mobile number to be update doesn't belongs to other user
+        const profile = Profile.findBy("mobileNumber", mobileNumber);
+        if (profile !== null) {
+          throw new ResourceExistError("Mobile number already used");
+        }
+      }else{
+        //Updating user profile
+        exitsedProfile.name = name;
+        exitsedProfile.gender = gender;
+        exitsedProfile.dateOfBirth = dateOfBirth;
+        exitsedProfile.mobileNumber = mobileNumber;
+        const updatedProfile = await exitsedProfile.save();
+        response.ok(updatedProfile);  
+      }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  //For deleting the user and profile based on mobile number
+  public async deleteUserProfile({ request, response, auth }) {
+    try {
+      const mobileNumber = request.input("mobileNumber");
+
+      //checking mobile number provide by user
+      if (mobileNumber === undefined) {
+        throw new InvalidDataError("Mobile number required");
+      }else{
+        const validMobileNumber = /^[0-9]{10}$/.test(mobileNumber);
+
+        if(validMobileNumber === true){
```

Comment:
handle this using validator.
For mobile format validation use `mobile` validator of adonis